src/main/scala/de/hpi/epic/pricewars/connectors/ProducerConnector.scala

```diff
@@ -65,7 +65,7 @@ object ProducerConnector {
     val producer_infos = encrypted_signature.split(" ")
 
     try {
-      if (producer_infos{0}.toLong == uid && producer_infos{1}.toInt == amount && producer_infos{2} == merchant_id) {
```

Comment:
What is your reason to not check for `producer_infos{1}.toInt >= amount`?
Is it sufficiently checked that no merchant sells more items than actually received from the producer?

Reply:
We could check for that, but this is implicitly done just a few lines below in line 71. This is also where it is checked that no merchant sells more items than received from the producer. If one tried to do that, `totalAmountUsed` would be greater than `producer_infos{1}.toInt` and the function would return `false`, hence the signature would not be correct.